janus_core/single_point.py

```diff
@@ -86,14 +102,29 @@ def __init__(
         calc_kwargs : Optional[dict[str, Any]]
             Keyword arguments to pass to the selected calculator. Default is {}.
         """
-        self.architecture = architecture
-        self.device = device
-        self.structure = structure
+        if struct and struct_path:
+            raise ValueError("Only one of `struct` and `struct_path` may be specified")
```

Comment:
maybe be more specific in here about what you mean by structure, ase Atoms object.